### DISCUSSION TITLE
Warum sind 401/2/3/4 und 2351 als O3EInt8 und nicht als O3EByteVal definiert?

### DISCUSSION THREAD

**Discussion by wangnick** [open3e/open3e#125] [Q&A] [answered]
Diese DIDs scheinen Zustände zu enumerieren und sind keine echten numerischen Werte.

**Comment by MyHomeMyData** [comment]
Diese DIDs sind alle nicht stimmig. 401-404 haben 5 Bytes, bisher werden nur 1 bzw. 2 Bytes ausgewertet. 405-408 sind dazu auch nicht konsistent.
Solange wir nicht mehr wissen, würde ich für **401-408** folgendes vorschlagen:
```
401 : O3EComplexType(5, "MixerOneCircuitPump", [RawCodec(2, "Unknown1"), O3EByteVal(1, "Mode"), RawCodec(2, "Unknown2")])
```

Und für **2351**:
```
2351 : O3EComplexType(2, "HeatPumpCompressor", [O3EByteVal(1, "Mode"), RawCodec(1, "Unknown")])
```
Vermutlich gilt das dann auch für **2352**:
```
2352 : O3EComplexType(2, "AdditionalElectricHeater", [O3EByteVal(1, "Mode"), RawCodec(1, "Unknown")])
```
**Hat jemand weitere Daten zu diesen DIDs oder andere Vorschläge?**

**Comment by MyHomeMyData** [comment]
Ich habe noch Infos von anderer Stelle bekommen. Damit stellt sich das nun so dar:
```
401 : O3EComplexType(5, "MixerOneCircuitPump", [O3EByteVal(1, "TargetPowerState"), O3EInt8(1,"TargetValue"), O3EByteVal(1, "PowerState"), O3EByteVal(1, "ErrorState"), O3EInt8(1,"ActualValue")])
2351 : O3EComplexType(2, "HeatPumpCompressor", [O3EByteVal(1, "PowerState"), O3EByteVal(1, "ErrorState")]),
2352 : O3EComplexType(2, "AdditionalElectricHeater", [O3EByteVal(1, "PowerState"), O3EByteVal(1, "ErrorState")]),
```

Jemand Einwände? Sonst stelle ich das so nach develop.

**Reply by MyHomeMyData** [reply, marked as answer]
In develop verfügbar.
